src/pages/StepEditor/StepEditor.tsx

```diff
@@ -32,36 +31,41 @@ function StepEditor() {
     enabled: !!stepId,
   });
 
-  const tabsData = [
-    { id: "1", title: "Step Settings", component: <StepSettings /> },
-    {
-      id: "2",
-      title: "Block Settings",
-      component: <BlockEditor stepId={stepId ?? ""} />,
-    },
-    { id: "3", title: "Data" },
-  ];
+  const addBlock = (blockType: BlockType) => {
+    const newBlock = {
+      quizId: quizId ?? '',
+      stepId: stepId ?? '',
+      type: blockType,
+      data: blockLibrary[blockType].factory(),
+    };
+
+    BlockClient.createBlock(newBlock);
+  };
 
   return (
-    <div>
+    <>
       <StepEditorProvider stepId={stepId}>
-        <>
-          <StepPreview step={step} quizId={quizId ?? ""} />
+        <Box w="100%" p={6}>
+          <Flex>
+            <PopoverModal header="Select block type:">
+              <Flex alignItems="center" justify="space-around" gap="2">
+                <Button size="sm" onClick={() => addBlock(BlockType.INPUT)}>
+                  {BlockType.INPUT}
+                </Button>
+                <Button size="sm" onClick={() => addBlock(BlockType.TEXTAREA)}>
+                  {BlockType.TEXTAREA}
+                </Button>
+              </Flex>
+            </PopoverModal>
```

Comment:
This should be the block inserter. (within the PopoverModal (or PopoverModalButton)